File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/server/command/vanilla/CommandSetWorldSpawn.java
package pize.tests.voxelgame.server.command.vanilla;

import pize.math.vecmath.vector.Vec3f;
import pize.tests.voxelgame.main.text.Component;
import pize.tests.voxelgame.main.text.TextColor;
import pize.tests.voxelgame.main.command.CommandContext;
import pize.tests.voxelgame.server.command.CommandDispatcher;
import pize.tests.voxelgame.main.command.builder.Commands;
import pize.tests.voxelgame.server.level.ServerLevel;
import pize.tests.voxelgame.server.player.ServerPlayer;

public class CommandSetWorldSpawn{
    
    public static void registerTo(CommandDispatcher dispatcher){
        dispatcher.newCommand(Commands.literal("setworldspawn")
            .requiresPlayer()
            .executes( CommandSetWorldSpawn::setWorldSpawn)
        );
    }
    
    private static void setWorldSpawn(CommandContext context){
        // Player
        final ServerPlayer sender = context.getSource().asPlayer();
        // Spawn position
        final Vec3f position = sender.getPosition();
        final ServerLevel level = sender.getLevel();
        // Set world spawn
        level.getConfiguration().setWorldSpawn(position.x, position.z);
        context.getServer().getPlayerList().broadcastServerMessage(new Component().color(TextColor.GREEN).text("World spawn set in: " + position));
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/server/command/vanilla/CommandSetWorldSpawn.kt
package pize.tests.voxelgame.server.command.vanilla

import pize.tests.voxelgame.main.command.CommandContext
import pize.tests.voxelgame.main.command.builder.Commands
import pize.tests.voxelgame.main.text.Component
import pize.tests.voxelgame.main.text.TextColor
import pize.tests.voxelgame.server.command.CommandDispatcher

object CommandSetWorldSpawn {
    fun registerTo(dispatcher: CommandDispatcher) {
        dispatcher.newCommand(
            Commands.literal("setworldspawn")
                .requiresPlayer()
                .executes { obj: CommandContext? -> setWorldSpawn() }
        )
    }

    private fun setWorldSpawn(context: CommandContext) {
        // Player
        val sender = context.source.asPlayer()
        // Spawn position
        val position = sender!!.position
        val level = sender.level
        // Set world spawn
        level.configuration.setWorldSpawn(position.x.toDouble(), position.z.toDouble())
        context.server.playerList.broadcastServerMessage(
            Component().color(TextColor.GREEN).text(
                "World spawn set in: $position"
            )
        )
    }
}
